src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonStringEnumConverter.cs

```diff
@@ -56,7 +58,13 @@ public sealed override bool CanConvert(Type typeToConvert)
         }
 
         /// <inheritdoc />
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
```

Comment:
Would it be sufficient to place the RDC attribute on this method? If so, it seems we'd be able to avoid the pragma suppressions in the generated code.

Reply:
It's a virtual method on JsonConverterFactory, so we need to annotate the constructor unfortunately.